homeassistant/components/homee/__init__.py

```diff
@@ -0,0 +1,87 @@
+"""The Homee integration."""
+
+import logging
+
+from pyHomee import Homee
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [Platform.COVER]
+
+type HomeeConfigEntry = ConfigEntry[Homee]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: HomeeConfigEntry) -> bool:
+    """Set up homee from a config entry."""
+    # Create the Homee api object using host, user,
+    # password & pyHomee instance from the config
+    homee = Homee(
+        host=entry.data[CONF_HOST],
+        user=entry.data[CONF_USERNAME],
+        password=entry.data[CONF_PASSWORD],
+        device="pymee_" + hass.config.location_name,
+        reconnect_interval=10,
+        max_retries=100,
+    )
+
+    # Start the homee websocket connection as a new task
+    # and wait until we are connected
+    hass.loop.create_task(homee.run())
+    await homee.wait_until_connected()
+
+    # Log info about nodes, to facilitate recognition of unknown nodes.
+    for node in homee.nodes:
+        _LOGGER.info(
+            "Found node %s, with following Data: %s",
+            node.name,
+            node.raw_data,
+        )
+
+    entry.runtime_data = homee
+
+    # create device register entry
+    device_registry = dr.async_get(hass)
+    device_registry.async_get_or_create(
+        config_entry_id=entry.entry_id,
+        connections={
+            (dr.CONNECTION_NETWORK_MAC, dr.format_mac(homee.settings.mac_address))
+        },
+        identifiers={(DOMAIN, homee.settings.uid)},
+        manufacturer="homee",
+        name=homee.settings.homee_name,
+        model="homee",
+        sw_version=homee.settings.version,
+    )
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    entry.async_on_unload(entry.add_update_listener(async_update_entry))
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: HomeeConfigEntry) -> bool:
+    """Unload a homee config entry."""
+    # Unload platforms
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+    if unload_ok:
+        # Get Homee object and remove it from data
+        homee: Homee = entry.runtime_data
+
+        # Schedule homee disconnect
+        homee.disconnect()
```

Comment:
You can also add `entry.async_on_unload(homee.disconnect)` to the setup code and it will automatically run when unloaded so this function can be clean

Reply:
That might also solve a current problem when running the tests. :-)